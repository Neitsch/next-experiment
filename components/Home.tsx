import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import gql from "graphql-tag";
import React from "react";

import { HomeFragment } from "../query-types/HomeFragment";
import ChangeUsername from "./ChangeUsername";

export const USER_QUERY = gql`
  fragment UserFragment on User {
    username
    id
  }
`;

export const HOME_QUERY = gql`
  fragment HomeFragment on Query {
    user {
      ...UserFragment
    }
  }
  ${USER_QUERY}
`;

export default class Home extends React.Component<{ data: HomeFragment }> {
  public render() {
    return (
      <Card>
        <CardContent>
          <div>
            {this.props.data.user.id} -{" "}
            {this.props.data.user.username
              ? this.props.data.user.username
              : "Anonymous"}
          </div>
          <div>
            <ChangeUsername />
          </div>
        </CardContent>
      </Card>
    );
  }
}
