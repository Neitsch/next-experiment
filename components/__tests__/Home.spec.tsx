import { shallow } from "enzyme";
import React from "react";

import Home from "../Home";

jest.unmock("../Home");

describe("Home", () => {
  describe("renders", () => {
    it("username", () => {
      const shallowRender = shallow(
        <Home data={{ user: { id: "test-id", username: "Test User" } }} />,
      );
      expect(shallowRender).toMatchSnapshot();
    });
    it("no username", () => {
      const shallowRender = shallow(
        <Home data={{ user: { id: "test-id" } }} />,
      );
      expect(shallowRender).toMatchSnapshot();
    });
  });
});
