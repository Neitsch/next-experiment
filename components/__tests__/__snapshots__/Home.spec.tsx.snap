// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders no username 1`] = `
<WithStyles(Card)>
  <WithStyles(CardContent)>
    <div>
      test-id
       -
       
      Anonymous
    </div>
    <div>
      <ChangeUsernameComponent />
    </div>
  </WithStyles(CardContent)>
</WithStyles(Card)>
`;

exports[`Home renders username 1`] = `
<WithStyles(Card)>
  <WithStyles(CardContent)>
    <div>
      test-id
       -
       
      Test User
    </div>
    <div>
      <ChangeUsernameComponent />
    </div>
  </WithStyles(CardContent)>
</WithStyles(Card)>
`;
